src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -1043,3 +1042,19 @@
   [c :- CertOrCSR
    digest :- schema/Str]
   (SSLUtils/getFingerprint c digest))
+
+(schema/defn ^:always-validate prevent-duplicate :- (schema/maybe '(schema/Int))
+  "Given the list of serials to be added to the revoked list of Puppet's CA CRL
+   and the Puppet's CA CRL, extract the all the existing serial numbers
+   within the Puppet's CA CRL and group them into a list. Compare the given list
+   of serials against the newly extracted list for any shared serial numbers.
+   Those shared serial numbers will be removed. Return a list of serials to be
+   added to the revoked list without any numbers already existed in Puppet's CA CRL."
```

Comment:
I think this could be simplified/summarized. You don't need to describe the details of the implementation, I think it's enough to just say something like,
"Given a list of serials and Puppet's CA CRL, returns the list of serials with any already-revoked serials removed."